### PR TITLE
implementing multipart upload

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: post-to-aws/src/main/java/no/sikt/nva/brage/migration/aws/MultiPartUploader.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    private CompleteMultipartUploadResponse multiPartUploadUsingClient(S3Client s3Client) throws IOException {
+        var request = initiateRequest();
+        var response = s3Client.createMultipartUpload(request);
+
+        int bytesRead;
+
+        try (var inputStream = new BufferedInputStream(new FileInputStream(file))) {
+            while ((bytesRead = readToBuffer(inputStream)) > 0) {
```
Comment (brinxmat) [inline post-to-aws/src/main/java/no/sikt/nva/brage/migration/aws/MultiPartUploader.java:64]: psf int EOS = 0;

File: post-to-aws/src/test/java/S3StorageImplTest.java [hunk 2]
```diff
@@ -145,4 +151,33 @@ var record = new Record();
...
+            return CreateMultipartUploadResponse.builder().uploadId(randomString()).build();
+        }
+
+        @Override
+        public UploadPartResponse uploadPart(UploadPartRequest request, RequestBody requestBody) {
+            return UploadPartResponse.builder().eTag(randomString()).build();
+        }
+
+        public String getSingleItem() {
+            return keys.get(0);
```
Comment (brinxmat) [inline post-to-aws/src/test/java/S3StorageImplTest.java:180]: Not on modern Java?

**Review by brinxmat — APPROVED** [review]
Two small comments

File: post-to-aws/src/test/java/S3StorageImplTest.java (hunk 2 above)
Comment (truhacevkir) [reply post-to-aws/src/test/java/S3StorageImplTest.java:180]: Nope, Java11, the version we have installed on all brage instances

**Review by anetteOlli — APPROVED** [review]

**Comment by anetteOlli** [comment]
Bra jobba.

**Pull request merged**